internal/ntt/fs.go

```diff
@@ -59,3 +61,47 @@ func findFiles(dir string, matcher func(name string) bool) ([]string, error) {
 
 	return sources, nil
 }
+
+func findAuxiliaryDirectories() []string {
+	var path string
+	var ret []string
+	var err error = nil
+	if path, err = exec.LookPath("k3r"); err != nil {
+		return nil
+	}
+	path = filepath.Dir(path)
+	var dirSuffix = []string{"/../lib/k3/plugins/ttcn3", "/../lib64/k3/plugins/ttcn3", "/../lib/x86_64/k3/plugins/ttcn3", "/../share/k3/ttcn3"}
+	pluginDirFound := false
+	for idx, realPath := range dirSuffix {
+		var finfo os.FileInfo = nil
+		realPath = path + realPath
+		if finfo, err = os.Stat(realPath); err != nil {
+			continue
+		}
+		if idx <= 2 {
+			if pluginDirFound {
+				continue
+			}
+			if finfo.IsDir() {
+				ret = append(ret, realPath)
+				pluginDirFound = true
+			}
+		} else {
+			if finfo.IsDir() {
+				ret = append(ret, realPath)
+			}
+		}
```

Comment:
I have some trouble understanding the function of variable `pluginDirFound`. Wouldn't it be valid to just return the first found directory?

```
if info, err := os.Stat(realPath); err == nil {
        if info.IsDir() {
                  return realPath
         }
}
```
If so, the signature of `findAuxiliaryDirectories` should also be changed to `func findAuxilaryDirectory() string`

Reply:
only for the first 3 entries. The last one is a directory with different content